File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    answer = 0
    for x in range(n+1):
        if x % 2 == 0 and n % 2 == 0:
            answer += (x**2)
        elif x % 2 == 1 and n % 2 == 1:
            answer += x
    return answer

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        boolean mode;
        
        if (n % 2 == 0){
            mode = true;
        }else {
            mode = false;           
        }
        
        for(int i = 1; i<=n; i++){
            if (mode) {
                if (i % 2 ==0){answer += i*i;}
            }else{
                if (i % 2 ==1){answer += i;}
            }
        }
        
        return answer;
    }
}